packages/toolbars-v3/web/src/components/SizeCalculator/SizeCalculator.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useState, useRef, useEffect } from 'react';
+
+type SizeResult = {
+  visible: React.Component[];
+  overflowed: React.Component[];
+};
+
+type SizeCalculatorProps = {
+  width: number;
+  components: React.Component[];
+  children: (result: SizeResult) => unknown;
+};
+
+const Overflow = ({
+  width,
+  components,
+  onChange,
+}: {
+  width: number;
+  components: React.Component[];
+  onChange;
+}) => {
+  const containerRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    let sumVisible = 0;
+    const visible: React.Component[] = [];
+    const overflowed: React.Component[] = [];
+
+    if (containerRef.current) {
+      const elements = Array.from(containerRef.current.children).map((child: HTMLElement) => {
+        return {
+          child,
+          width: Math.ceil(child.getBoundingClientRect().width),
+        };
+      });
+
+      elements.forEach((element, index) => {
+        if (sumVisible + element.width <= width) {
+          visible.push(components[index]);
+          sumVisible += element.width;
+        } else {
+          overflowed.push(components[index]);
+        }
+      });
```

Comment:
```suggestion
      elements.forEach((element, index) => {
        if (sumVisible + element.width <= width) {
          visible.push(components[index]);
        } else {
          overflowed.push(components[index]);
        }
        sumVisible += element.width;
      });
```
there can be a strange behavior without this change (I know from my PR)